template_feed/Microsoft.DotNet.Web.ProjectTemplates.2.0/content/RazorPagesWeb-CSharp/Pages/Account/Manage/Disable2fa.cshtml.cs

```diff
@@ -28,12 +28,12 @@ public class Disable2faModel : PageModel
             var user = await _userManager.GetUserAsync(User);
             if (user == null)
             {
-                throw new ApplicationException($"Unable to load user with name '{User.Identity.Name}'.");
+                throw new ApplicationException($"Unable to load user with id '{_userManager.GetUserId(User)}'.");
```

Comment:
A better error message for all of the GetUserAsync failures, might be more along the lines of, "User is not signed in"